### ISSUE TITLE
Confirm appropriate MAXMAG is set on the MATLAB side

### ISSUE THREAD

**Issue by jeanconn** [sot/proseco#121]

**Comment by jeanconn** [comment]
maxmag set here

https://github.com/sot/fot_matlab_tools/blob/review_4.0/MissionScheduling/stars/%40ACACatalog/private/convertStarSelectorClass.m#L63

New code is setting this via this:

https://github.com/sot/fot_matlab_tools/blob/bac55a7f9695bd36599e4ddca0cb04ced435c4ae/MissionScheduling/stars/StarSelector/%40StarSelector/private/assignProsecoCatalog.m#L19

Previously set here

https://github.com/sot/fot_matlab_tools/blob/bac55a7f9695bd36599e4ddca0cb04ced435c4ae/MissionScheduling/stars/StarSelector/%40StarSelector/private/getMagAndMagError.m#L10

**Comment by jeanconn** [comment]
From my previous notes on this, my understanding is the issue is that the MAXMAG - MAG values were set as a side-effect of SAUSAGE star selection.  The trick is that the SAUSAGE implementation of this effectively could end up with different values for MAXMAG for a single star depending on in which stage the star was selected (different clipping by stage, actually decreasing MAXMAG - MAG in later stages).

This isn't reproduced in the proseco/Matlab implementation because proseco does not provide those stages or work that way.

I also didn't just override MAXMAG to be MAG + 1.5 in the combined proseco/matlab implementation because I didn't get SSAWG to agree to that choice.

Instead, there's somewhat the worst-case compromise, where, for stars with (MAG_ACA_ERR > ~0.5  mag and MAG_ACA_ERR < 1.0 mag), MAXMAG - MAG mag may be 3 * the error term and not just 1.5. 
Stars with MAG_ACA_ERR > 1.0 mag are not selected.  Stars with MAG_ACA_ERR < ~0.5 will just end up with the error term being 0.5 and MAXMAG - MAG == 1.5 .

It would be my preference to just set MAXMAG - MAG to be 1.5 or less in the future.

For testing, the test plan described by Tom seems reasonable; we should compare MAXMAG values in the test products to the current release version of the tools and sanity check.

**Comment by taldcroft** [comment]
The validate-orviewer-maxmag.ipynb notebook in the usual MTA aspect proseco www area does this quick sanity check.  Looks good to me.

**Comment by taldcroft** [comment]
@jeanconn - nice summary of the situation, and this matches what I saw in products.  (Though oddly the factor is a bit more than 3, so there is some other term being added in).

You should copy this whole explanation out to a new top-level issue so we can track it since this issue will (hopefully) be closed soon.

**Comment by jeanconn** [comment]
What do we need as a top-level issue (I thought this was the top level issue)?

Also, when I said "MAXMAG - MAG mag may be 3 * the error term", the error term includes MAG_ACA_ERR but is not just MAG_ACA_ERR (I think the "random term" of 0.26 is added in quadrature).

**Comment by taldcroft** [comment]
This issue is narrowly defined for "4.3.x final release" as confirming that MATLAB is matching expectation for the *current* algorithm.

I've updated https://github.com/sot/ska-projects/issues/38 to include your discussion and give that issue a more appropriate title.

**Comment by jeanconn** [comment]
I figured there is a remaining action to put a summary these notes and the results of the notebook in the presentation.

**Comment by taldcroft** [comment]
This is too low level for the presentation.  Let me know when you have reviewed the notebook.

**Comment by jeanconn** [comment]
Right, I suppose I meant that in addition to the document describing the guide algorithm, there should be something somewhere that notes there are some gotchas in the matlab/proseco integration?  Or is this issue that in full?

**Comment by jeanconn** [comment]
Regarding the notebook, do we need to compare again to the previous behavior or do we just not care (as long as the tested behavior is reasonable the differences don't matter)?
